x-pack/plugins/cases/server/client/cases/bulk_update.ts

```diff
@@ -348,12 +348,22 @@ export const bulkUpdate = async (
       casesMap,
       query.cases
     );
-
-    await authorization.ensureAuthorized({
-      entities: casesToAuthorize,
-      operation: Operations.updateCase,
+    const requestToChangeStatusFromClosed = Array.from(casesMap.values()).some((c) => {
+      return c.attributes.status === CaseStatuses.closed;
     });
 
+    if (requestToChangeStatusFromClosed) {
+      await authorization.ensureAuthorized({
+        entities: casesToAuthorize,
+        operation: Operations.reopenCases,
```

Comment:
Thoughts on the following? Keep `ensureAuthorized` as simple as possible rather than accounting for arrays vs singular operations?


```
      const operationsToAuthorized = requestToChangeStatusFromClosed
        ? [Operations.reopenCase, Operations.updateCase]
        : [Operations.updateCase];

      await Promise.all(
        operationsToAuthorized.map(async (operationToAuthorize) =>
          authorization.ensureAuthorized({
            entities: casesToAuthorize,
            operation: operationToAuthorize,
          })
        )
      );
```

Reply:
I would suggest making the changes in the `ensureAuthorized` to support arrays as we want to authorize the user for both operations at the same time.  Probably authorizing each action one at a time instead of authorizing both at the same time is the same but I am not 100% sure. Also, I would like to avoid weird race conditions that can lead to bugs that would be difficult to catch and debug. The kibana security privileges already support multiple actions (operations) at the moment. It should be straightforward. We can pair if you want.